assembly/src/library/version.rs

```diff
@@ -0,0 +1,173 @@
+use core::{
+    fmt,
+    str::{self, FromStr},
+};
+
+use crate::{
+    diagnostics::Diagnostic, ByteReader, ByteWriter, Deserializable, DeserializationError,
+    Serializable,
+};
+
+/// Represents a _Semantic Versioning_ version string, without pre-releases.
+#[derive(Debug, Copy, Clone, Eq, PartialEq, PartialOrd, Ord, Hash)]
+pub struct Version {
+    /// The major version, incremented when breaking changes occur.
+    pub major: u16,
+    /// The minor version, incremented when new features or functionality is introduced.
+    pub minor: u16,
+    /// The patch version, incremented when non-breaking changes are made.
+    pub patch: u16,
+}
+
+/// Construction
+impl Version {
+    /// Returns the current minimal version supported by the code in this crate.
+    #[inline(always)]
+    pub const fn min() -> Self {
+        Self {
+            major: 1,
+            minor: 0,
+            patch: 0,
+        }
+    }
+}
```

Comment:
This could be `0.2.0` instead, for sure. For context, I wanted primarily to indicate that a breaking change had occurred, so I bumped the major version. However, going from a major of 0 to 1 _is_ saying "this format is stable now", which may or may not be true at this point, so I think it probably doesn't make sense to do that yet. I don't expect the format is likely to change, but I also don't think MASL files will continue to exist once we make the switch to MAST (unless we decide to repurpose MASL as the name of our package format), so 🤷

Reply:
I think we are using this to indicate versions of specific libraries (rather than version of MASL format). For example, for `stdlib` this is set to `0.8.0`, I believe. And for `miden-lib`, this is still at `0.1.0`.

Reply:
Ah, I see. In that case, I'll revert this to `0.1.0` then.